app/src/pages/income.tsx

```diff
@@ -102,12 +106,30 @@ const TD = styled.td`
   min-width: 20rem;
 `
 
-export const getServerSideProps: GetServerSideProps = async ({ locale }) => {
-  return {
+export const getServerSideProps: GetServerSideProps = async ({
+  locale,
+  req,
+}) => {
+  const prevRouteIndex = req.headers.referer?.lastIndexOf('/')
+  const previousRoute =
+    prevRouteIndex && req.headers.referer?.substring(prevRouteIndex)
+  let returnval: GetServerSidePropsResult<{ [key: string]: object }> = {
     props: {
       ...(await serverSideTranslations(locale || 'en', ['common'])),
     },
   }
+
+  if (!['/eligibility', '/clinic'].includes(previousRoute as string)) {
```

Comment:
This might change to include `/review` depending on what Daniella says. Let's leave this for now.